File: LC-71-Simplify-Path/Solution.java
class Solution {
    public String simplifyPath(String path) {
        String[] sp = path.split("/");
        List<String> list = new ArrayList<String>();

        for(String s : sp){
            if(s.equals("") | s.equals("."))continue;
            else if(s.equals("..")){
                if(list.size() > 0){
                    list.remove(list.size()-1);
                }
            }
            else{
                list.add(s);
            }
        }

        StringBuilder sb = new StringBuilder();

        for(String s : list){
            sb.append("/");
            sb.append(s);
        }

        if(sb.length() == 0) return "/";

        return sb.toString();
    }
}

File: LC-71-Simplify-Path/Solution.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        sp = path.split("/")
        res = ""
        stack = [] 
        for i in sp:
            
            if i=="" or i == ".":
                continue
            elif i == "..":
                if len(stack) > 0 :
                    stack.pop()
            else:
                stack.append(i)
        
        for i in stack:
            res = res+"/"+i
        if len(res) == 0:
            return "/"
        return res
